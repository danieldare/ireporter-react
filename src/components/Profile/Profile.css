.profile-head {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 50px 0;
}

.profile-info {
  font-family: "Nunito";
}

.profile-img {
  border-radius: 50%;
  background: #d1603d;
  color: #fff;
  font-size: 5.5rem;
  display: inline-block;
  padding: 15px 25px;
  margin-right: 20px;
}

.types {
  padding: 20px 0;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}

.type-number {
  text-align: center;
  font-size: 3rem;
  border: 1px solid #ccc;
  display: inline-block;
  padding: 2px 10px;
  margin: 10px 0;
  border-radius: 5px;
}

.type-family {
  text-align: center;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
  padding: 20px 0;
}

.type-toggle {
  display: inline-block;
  padding: 10px 20px;
  border-radius: 1px;
  box-shadow: -1px 3px 5px rgba(0, 0, 0, 0.3);
  cursor: pointer;

  border: 1px solid rgb(200, 200, 200);
  transition: box-shadow 100ms ease-in-out;
}

.type-name {
  font-weight: 600;
}

.email-holder {
  margin: 5px 0;
}

.type-toggle.active {
  border: 1px solid rgb(190, 190, 190);
  box-shadow: none;
  font-weight: 800;
  color: #d1603d;
}

.buttin {
  padding: 5px;
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  background: transparent;
  border: 1px solid #ccc;
  font-size: 1.2rem;
  text-decoration: none;
}

.del {
  background: #d1603d;
  color: #fff;
  border: none;
}

.buttin:not(:last-child) {
  margin-right: 15px;
}

.table {
  width: 100%;
  border-collapse: collapse;
}

tr {
  width: 100%;
  border: 1px solid rgb(231, 231, 231);
  padding: 15px 0;
}

.th-head {
  box-shadow: 0 3px 8px rgba(0, 0, 0, 0.19);
  border: 1px solid rgb(212, 212, 212);
}

td,
th {
  border: 1px solid #dddddd;
  padding: 8px;
}

th:not(:last-child),
td:not(:last-child) {
  border-right: 1px solid #ccc;
}

.spinner-tr {
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
  align-items: center;
  padding: 20px;
}
