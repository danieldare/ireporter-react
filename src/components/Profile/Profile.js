import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import {
  getIncidents,
  deleteIncident
} from '../../actions/actionCreators/profileAction';

const Profile = props => {
  const {
    profileData: { incidents, isDeleting, isLoading }
  } = props;

  const [isType, setTypeState] = useState(false);
  useEffect(() => {
    props.getIncidents();
  }, []);

  const onDeleteClick = idNum => {
    incidents.map(({ type, id }) => {
      if (type === 'red-flag' && id === +idNum) {
        return props.deleteIncident(id, 'red-flags');
      }
      if (type === 'intervention' && id === +idNum) {
        return props.deleteIncident(id, 'interventions');
      }

      return '';
    });
  };

  let draftCount = 0;
  let investigationCount = 0;
  let resolvedCount = 0;
  let rejectedCount = 0;

  const dataTable = (
    <section className="table-container">
      <table className="table">
        <thead>
          <tr className="th-head">
            <th>S/N</th>
            <th>Title</th>
            <th>Type</th>
            <th>Action {isDeleting ? <div className="spinner-2" /> : ''}</th>
          </tr>
        </thead>

        {!isLoading ? (
          incidents.map(({ title, type, id, status }, index) => {
            if (type === 'red-flag' && isType) {
              if (status === 'resolved') {
                resolvedCount += 1;
              } else if (status === 'rejected') {
                rejectedCount += 1;
              } else if (status === 'under-investigation') {
                investigationCount += 1;
              } else {
                draftCount += 1;
              }
            } else if (type === 'intervention' && !isType) {
              if (status === 'resolved') {
                resolvedCount += 1;
              } else if (status === 'rejected') {
                rejectedCount += 1;
              } else if (status === 'under-investigation') {
                investigationCount += 1;
              } else {
                draftCount += 1;
              }
            }
            return (
              <tbody key={id}>
                <tr>
                  <td>{index + 1}</td>
                  <td>{title}</td>
                  <td>{type}</td>
                  <td>
                    <Link
                      to={
                        type === 'red-flag'
                          ? `/view-single-redflag?id=${id}`
                          : `/view-single-intervention?id=${id}`
                      }
                      className="buttin"
                    >
                      View
                    </Link>
                    <button
                      type="button"
                      className="buttin del"
                      onClick={() => onDeleteClick(id)}
                    >
                      Delete
                    </button>
                  </td>
                </tr>
              </tbody>
            );
          })
        ) : (
          <div className="spinner-tr">
            <div className="spinner-2" />
            <p>loading...</p>
          </div>
        )}
      </table>
      {isLoading === false && incidents.length < 1 ? (
        <p style={{ margin: '20px 0', textTransform: 'uppercase' }}>
          No incident found
        </p>
      ) : (
        ''
      )}
    </section>
  );

  return (
    <div className="container">
      <h2 className="user-info2">My Profile</h2>
      <div className="profile-head">
        <div className="profile-img">W</div>
        <div className="profile-info">
          <p className="email-holder">dodanieloluwadare@gmail.com</p>
          <p>Oluwadare Daniel Segun</p>
        </div>
      </div>
      <div className="type-family">
        <p
          className={`type-toggle ${isType ? ' active' : ' '}`}
          role="presentation"
          onClick={() => setTypeState(!isType)}
          onKeyDown={() => setTypeState(!isType)}
        >
          RedFlag
        </p>
        <p
          className={`type-toggle ${!isType ? ' active' : ' '}`}
          role="presentation"
          onClick={() => setTypeState(!isType)}
          onKeyDown={() => setTypeState(!isType)}
        >
          Intervention
        </p>
        <div className="types">
          <div className="type">
            <p className="type-number">{draftCount}</p>
            <p className="type-name">Draft</p>
          </div>
          <div className="type">
            <p className="type-number">{investigationCount}</p>
            <p className="type-name">Under-investigation</p>
          </div>
          <div className="type">
            <p className="type-number">{resolvedCount}</p>
            <p className="type-name">Resolved</p>
          </div>
          <div className="type">
            <p className="type-number">{rejectedCount}</p>
            <p className="type-name">Rejected</p>
          </div>
        </div>
        {dataTable}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  profileData: state.profile
});

export default connect(
  mapStateToProps,
  { getIncidents, deleteIncident }
)(withRouter(Profile));
