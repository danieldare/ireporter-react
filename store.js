import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './src/reducers';

const initialState = {};

const middlewares = [thunk];

if (process.env.NODE_ENV === `development`) {
  middlewares.push(createLogger());
}

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);

export default store;
