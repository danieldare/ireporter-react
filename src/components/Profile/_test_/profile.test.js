import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Profile from '../Profile';
import store from '../../../../store';
import mockRouterOptions from '../../../../test/__mocks__/mockRouter';

// describe('<App />', () => {
//   it('should match snapshot', () => {
//     const tree = shallow(<Profile />);
//     expect(tree).toMatchSnapshot();
//   });
// });

describe('<Profile />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
