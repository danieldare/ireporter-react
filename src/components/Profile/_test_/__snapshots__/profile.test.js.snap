// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "Records": Object {
          "errors": null,
          "incidents": Array [],
          "isFetching": false,
        },
        "auth": Object {
          "isAuthenticated": null,
          "isLoading": false,
          "token": null,
          "user": "",
        },
        "error": Object {
          "id": null,
          "msg": Object {},
          "status": null,
        },
        "post": Object {
          "incident": Object {},
          "incidents": Array [],
          "isCreating": false,
          "isLoading": false,
          "success": false,
        },
        "profile": Object {
          "error": null,
          "incidents": Array [],
          "isLoading": false,
        },
        "singleRecord": Object {
          "errors": null,
          "incident": "",
          "isFetching": false,
        },
      },
    }
  }
>
  <BrowserRouter>
    <Connect(withRouter(Profile)) />
  </BrowserRouter>
</ContextProvider>
`;
